dts/arm/nxp/nxp_k66.dtsi

```diff
@@ -34,6 +34,10 @@
 			sjw = <1>;
 			sample-point = <875>;
 			status = "disabled";
+			net_canbus1 {
+				compatible = "net-canbus";
```

Comment:
You are introducing a new DTS compatible here without adding a DTS binding. 

DTS is supposed to describe hardware, not software. I am not fond of describing what is basically a network stack on top of CAN via the devicetree.

Reply:
I have given this some more thought and will try to come up with an alternative solution, where the CAN driver instances are placed in an iteratable section similar to what was recently done for the sensor drivers. This will allow the network canbus driver to create a network interface for each CAN controller without having to add additional configuration to the dts.

Reply:
> 

Appreciate you looking into it!

Reply:
@henrikbrixandersen ..
Please evaluate this idea...
If I trigger socketcan init from CAN device init (let's say I add an API to trigger socketcan init in can_common.c or can_handler.c) (this choice of creating a socketcan interface can be based on a new property in CAN DT nodes)

Reply:
@henrikbrixandersen ..I have a few queries about putting CAN driver instances in ITERABLE_SECTION.
Q1. Can the order of CAN instances in this section vary? Since DT parsing order can change.
Q2. On what basis will network canbus driver choose its underlying CAN instance? (If this choice is based on the index of section entries, then it might change)

My point is that the correlation between the network canbus instance and the CAN instance should be consistent.

Reply:
> Q1. Can the order of CAN instances in this section vary? Since DT parsing order can change.

The instances in the iterable section will be sorted by the alphanumeric device name string.

> Q2. On what basis will network canbus driver choose its underlying CAN instance? (If this choice is based on the index of section entries, then it might change)

The canbus network driver will iterate the sorted sections in the order given by the alphanumeric device name string of the underlying CAN controller devices and create a canbus network interface for each.